File: src/net/pryoscode/CustomDiscordRPC/Discord.java
package net.pryoscode.CustomDiscordRPC;

import club.minnced.discord.rpc.DiscordEventHandlers;
import club.minnced.discord.rpc.DiscordRPC;
import club.minnced.discord.rpc.DiscordRichPresence;
import com.google.gson.Gson;
import com.google.gson.JsonObject;
import net.pryoscode.CustomDiscordRPC.utils.FileUtils;
import java.awt.*;
import java.util.Timer;
import java.util.TimerTask;

public class Discord {

    private long startTimestamp;
    private String clientID = "";

    public Discord() {
        DiscordRPC lib = DiscordRPC.INSTANCE;
        DiscordEventHandlers handlers = new DiscordEventHandlers();
        handlers.ready = (user) -> Main.trayUtils.getTrayIcon().displayMessage("CustomDiscordRPC " + Main.VERSION, "Successfully started the RPC Client.", TrayIcon.MessageType.INFO);
        startTimestamp = System.currentTimeMillis() / 1000;

        Timer timer = new Timer();
        timer.scheduleAtFixedRate(new TimerTask() {
            @Override
            public void run() {
                try {
                    Gson gson = new Gson();
                    JsonObject jsonObject = gson.fromJson(FileUtils.read(FileUtils.getConfig()), JsonObject.class);

                    if (!clientID.equals(jsonObject.get("ClientID").getAsString())) {
                        clientID = jsonObject.get("ClientID").getAsString();
                        lib.Discord_Shutdown();
                        lib.Discord_Initialize(clientID, handlers, true, "");
                    }

                    DiscordRichPresence richPresence = new DiscordRichPresence();
                    richPresence.startTimestamp = startTimestamp;

                    richPresence.state = jsonObject.get("State").getAsString();
                    richPresence.details = jsonObject.get("Details").getAsString();

                    JsonObject images = jsonObject.get("Images").getAsJsonObject();
                    richPresence.largeImageKey = images.get("LargeImage").getAsString();
                    richPresence.largeImageText = images.get("LargeImageTooltip").getAsString();
                    richPresence.smallImageKey = images.get("SmallImage").getAsString();
                    richPresence.smallImageText = images.get("SmallImageTooltip").getAsString();

                    lib.Discord_UpdatePresence(richPresence);
                    lib.Discord_RunCallbacks();
                } catch (Exception e) {
                    Main.errorPopup(e);
                }
            }
        }, 0, 2000);
    }

}

File: src/main/kotlin/net/pryoscode/customdiscordrpc/Discord.kt
package net.pryoscode.customdiscordrpc

import kotlin.io.println
import club.minnced.discord.rpc.DiscordRPC
import club.minnced.discord.rpc.DiscordEventHandlers
import club.minnced.discord.rpc.DiscordRichPresence
import club.minnced.discord.rpc.DiscordUser

class Discord : DiscordEventHandlers.OnReady {

    private val tray: SysTray

    constructor(tray: SysTray) {
        this.tray = tray
        val lib = DiscordRPC.INSTANCE
        val handlers = DiscordEventHandlers()
        handlers.ready = this
        val presence = DiscordRichPresence()
        presence.startTimestamp = System.currentTimeMillis() / 1000
        var clientId = ""
        Thread {
            while (!Thread.currentThread().isInterrupted()) {
                if (!clientId().equals(clientId)) {
                    clientId = clientId()
                    lib.Discord_Shutdown()
                    lib.Discord_Initialize(clientId, handlers, true, "")
                }
                presence.details = details()
                presence.state = state()
                presence.largeImageKey = imagesLargeKey()
                presence.largeImageText = imagesLargeText()
                presence.smallImageKey = imagesSmallKey()
                presence.smallImageText = imagesSmallText()
                lib.Discord_UpdatePresence(presence)
                lib.Discord_RunCallbacks()
                Thread.sleep(2000)
            }
        }.start()
    }

    override fun accept(user: DiscordUser) {
        tray.display("Successfully connected to Discord.")
    }

}